c3d/tests/basic0.h

```diff
@@ -1,17 +1,19 @@
 #include <stdint.h>
 
 typedef uint32_t FOO_ID;
+typedef uint16_t BAR_ID;
 
 #define FOO_MAJOR_VERSION 0
 #define FOO_MINOR_VERSION 0
 
+#define BAR_MAJOR_VERSION 123
+#define BAR_MINOR_VERSION 42
+
 typedef struct
-[[
-  everparse::entrypoint,
-  everparse::process,
-  everparse::parameter(uint32_t MessageBodyLength),
-  everparse::where(MessageBodyLength == sizeof(this))
-]]
+  [[everparse::process(1)]]
```

Comment:
I just needed some dummy expression, since I made process take an argument (to get notified of its parsing). Anything works and is ignored.

Reply:
why don't you just override handleDeclAttribute to i) call `TrivialC3dAttrInfo::handleDeclAttribute` (to resolve the virtual call properly) then ii) do whatever needs to be done once the attribute is processed?

Reply:
The problem is that I need to affect the parsing scope, and using `handleDeclAttribute` is too late, since afaict that only gets called after parsing all the attributes for struct. Consider
```c++
struct
  [[everparse::process,
    everparse::parameter(int a)]]
s1 {  ... };

struct
  [[everparse::process,
    everparse::where(a > 0)]]
s2 {  ... };
```
I need this callback to kick in at the time of parsing the second `everparse::process`, so the scope gets reset before parsing the `where` clause, lest it would succeed. Or maybe I misunderstood your suggestion..?

Reply:
sorry, I meant diagAppertainsToDecl which I think is called earlier

if the sequence is

diagAppertainsToDecl(process)
diagAppertainsToDecl(int a)
parseAttributePayload(int a)

then you could conceivably flip a flag in the `diagAppertainsToDecl(process)` that says "whoever gets access to the parser next resets the scope"? what do you think?

Reply:
Ah! I think it's not that way actually.. `diagAppertainsToDecl` receives a `ParsedAttr&` as argument, so it happens after parsing the argument. Also that's not interleaved, it seems all the parsing is done and then the `diagAppertainsToDecl` checks are made, so `diagAppertainsToDecl(process)` happens after `parseAttributePayload(where)`.

Reply:
Ah yes that order I describe is the one proposed for the next iteration of the clang patch, not the way things are currently done. Let's leave it as-is then, that should do for now. Thanks for the explanations, and perhaps worth a comment somewhere.

Note that with the future iteration of the patch, we'll be able to play that sort of tricks and make the argument optional.

Reply:
Ah! That's great. Sounds good.